### ISSUE TITLE
CA1861 FP for top-level statement

### ISSUE THREAD

**Issue by hankovich** [dotnet/roslyn-analyzers#7111]
### Analyzer

**Diagnostic ID**: CA1861 [Avoid constant arrays as arguments (code analysis)](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1861)

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 8.0.0 (Latest)

### Describe the bug

1. CA1861 is reported in top-level statements, but I can not create a `static readonly` field in top-level statements
2. The exception is thrown when I try to apply a code fix.

### Steps To Reproduce

```csharp
_ = "a".Split(new[] { "1", "2" }, StringSplitOptions.None);
```

### Expected behavior

Either
1. No warnings are reported
2. Code fix fixes reported warning

## Additional context

```
System.InvalidOperationException: The item specified is not the element of a list.
   at Microsoft.CodeAnalysis.CSharp.Syntax.SyntaxReplacer.NodeListEditor.Visit(SyntaxNode node) in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Compilers\CSharp\Portable\Syntax\SyntaxReplacer.cs:line 388
   at Microsoft.CodeAnalysis.CSharp.Syntax.SyntaxReplacer.InsertNodeInList(SyntaxNode root, SyntaxNode nodeInList, IEnumerable`1 nodesToInsert, Boolean insertBefore) in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Compilers\CSharp\Portable\Syntax\SyntaxReplacer.cs:line 242
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode.InsertNodesInListCore(SyntaxNode nodeInList, IEnumerable`1 nodesToInsert, Boolean insertBefore) in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Compilers\CSharp\Portable\Syntax\CSharpSyntaxNode.cs:line 500
   at Microsoft.CodeAnalysis.SyntaxNodeExtensions.InsertNodesBefore[TRoot](TRoot root, SyntaxNode nodeInList, IEnumerable`1 newNodes) in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Compilers\Core\Portable\Syntax\SyntaxNodeExtensions.cs:line 104
   at Microsoft.CodeAnalysis.Editing.SyntaxGenerator.InsertNodesBefore(SyntaxNode root, SyntaxNode node, IEnumerable`1 newDeclarations) in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Workspaces\Core\Portable\Editing\SyntaxGenerator.cs:line 1388
   at Microsoft.CodeAnalysis.CSharp.CodeGeneration.CSharpSyntaxGenerator.InsertNodesBeforeInternal(SyntaxNode root, SyntaxNode declaration, IEnumerable`1 newDeclarations) in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Workspaces\CSharp\Portable\CodeGeneration\CSharpSyntaxGenerator.cs:line 2902
   at Microsoft.CodeAnalysis.CSharp.CodeGeneration.CSharpSyntaxGenerator.
c__DisplayClass207_0. b__1(SyntaxNode r) in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Workspaces\CSharp\Portable\CodeGeneration\CSharpSyntaxGenerator.cs:line 2889
   at Microsoft.CodeAnalysis.Editing.SyntaxGenerator.PreserveTrivia[TNode](TNode node, Func`2 nodeChanger) in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Workspaces\Core\Portable\Editing\SyntaxGenerator.cs:line 1469
   at Microsoft.CodeAnalysis.CSharp.CodeGeneration.CSharpSyntaxGenerator.Isolate(SyntaxNode declaration, Func`2 editor) in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Workspaces\CSharp\Portable\CodeGeneration\CSharpSyntaxGenerator.cs:line 2006
   at Microsoft.CodeAnalysis.CSharp.CodeGeneration.CSharpSyntaxGenerator.InsertNodesBefore(SyntaxNode root, SyntaxNode declaration, IEnumerable`1 newDeclarations) in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Workspaces\CSharp\Portable\CodeGeneration\CSharpSyntaxGenerator.cs:line 2889
   at Microsoft.CodeAnalysis.Editing.SyntaxEditor.InsertChange.Apply(SyntaxNode root, SyntaxNode currentNode, SyntaxGenerator generator) in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Workspaces\Core\Portable\Editing\SyntaxEditor.cs:line 317
   at Microsoft.CodeAnalysis.Editing.SyntaxEditor.Change.Apply(SyntaxNode root, SyntaxGenerator generator) in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Workspaces\Core\Portable\Editing\SyntaxEditor.cs:line 257
   at Microsoft.CodeAnalysis.Editing.SyntaxEditor.GetChangedRoot() in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Workspaces\Core\Portable\Editing\SyntaxEditor.cs:line 112
   at Microsoft.NetCore.Analyzers.Runtime.AvoidConstArraysFixer.ExtractConstArrayAsync(Document document, SyntaxNode root, SyntaxNode node, ImmutableDictionary`2 properties, CancellationToken cancellationToken)
   at Microsoft.NetCore.Analyzers.Runtime.AvoidConstArraysFixer.
c__DisplayClass5_0. b__0>d.MoveNext() 
   --- End of stack trace from previous location ---
   at Microsoft.CodeAnalysis.CodeActions.CodeAction.GetChangedSolutionAsync(CancellationToken cancellationToken) in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Workspaces\Core\Portable\CodeActions\CodeAction.cs:line 219
   at Microsoft.CodeAnalysis.CodeActions.CodeAction.ComputeOperationsAsync(CancellationToken cancellationToken) in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Workspaces\Core\Portable\CodeActions\CodeAction.cs:line 190
   at Microsoft.CodeAnalysis.CodeActions.CodeAction.ComputeOperationsAsync(IProgressTracker progressTracker, CancellationToken cancellationToken) in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Workspaces\Core\Portable\CodeActions\CodeAction.cs:line 202
   at Microsoft.CodeAnalysis.CodeActions.CodeAction.GetOperationsCoreAsync(Solution originalSolution, IProgressTracker progressTracker, CancellationToken cancellationToken) in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Workspaces\Core\Portable\CodeActions\CodeAction.cs:line 154
   at JetBrains.Roslyn.Worker.Workspaces.Analyzers.CodeActionHelper.GetOperationsSafeAsync(CodeAction codeAction, CancellationToken cancellationToken)
```

**Comment by Youssef1313** [comment]
The codefix should create a `partial class Program` and add the static field in it
